src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityCharger.java

```diff
@@ -87,18 +90,274 @@ protected void initializeInventory() {
         this.itemInventory = importItems;
     }
 
+    @Override
+    public void update() {
+        super.update();
+        if (!getWorld().isRemote) {
+            this.batterySlotsUsedThisTick.clear();
+            if(getEnergyStored() > 0) {
+                chargePlayerInRegion();
+
+                //Charge the internal batteries
+                for (int i = 0; i < importItems.getSlots(); i++) {
+                    ItemStack batteryStack = importItems.getStackInSlot(i);
+                    IElectricItem electricItem = getBatteryContainer(batteryStack);
+
+                    if (electricItem != null) {
+                        long inputVoltage = Math.min(energyContainer.getInputVoltage(), energyContainer.getEnergyStored());
+                        long energyUsed = electricItem.charge(inputVoltage, getTier(), false, false);
+                        if(energyUsed > 0L) {
+                            energyContainer.removeEnergy(energyUsed);
+                            importItems.setStackInSlot(i, batteryStack);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Finds players within a 2 block region of the front face of the machine and charges items in their inventory
+     */
+    private void chargePlayerInRegion() {
+
+        BlockPos endPos;
+        //If not facing up or down, add an additional translation, to translate the charging region to be centered on a block,
+        //rather than centered between blocks
+        if (!(frontFacing == EnumFacing.UP || frontFacing == EnumFacing.DOWN)) {
+            endPos = this.getPos().offset(frontFacing, 2).add(0, 1, 0).add(0, 0, 1);
+        } else {
+            endPos = this.getPos().offset(frontFacing, 2);
+        }
+        //Create a list of all players within the AABB created by the charge position and the end position of the charging region
+        List<EntityPlayer> playersAbove = this.getWorld().getEntitiesWithinAABB(EntityPlayer.class, new AxisAlignedBB(this.getPos(), endPos));
```

Comment:
I think you need to use Predicate version of the getEntitiesWithinAABB() here to check to see if the player is dead?

The player entity can still exist until the user clicks "respawn".